/*
 * This file was automatically generated by EvoSuite
 * Mon Apr 08 02:23:25 GMT 2024
 */

package org.myrobotlab.document.transformer;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import org.myrobotlab.document.Document;
import org.myrobotlab.document.transformer.MathValues;
import org.myrobotlab.document.transformer.StageConfiguration;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class MathValues_ESTest extends MathValues_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      MathValues mathValues0 = new MathValues();
      StageConfiguration stageConfiguration0 = new StageConfiguration();
      // Undeclared exception!
      try { 
        mathValues0.startStage(stageConfiguration0);
        fail("Expecting exception: NoClassDefFoundError");
      
      } catch(NoClassDefFoundError e) {
         //
         // net/objecthunter/exp4j/ExpressionBuilder
         //
         verifyException("org.myrobotlab.document.transformer.MathValues", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      MathValues mathValues0 = new MathValues();
      // Undeclared exception!
      try { 
        mathValues0.processDocument((Document) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.myrobotlab.document.transformer.MathValues", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      MathValues mathValues0 = new MathValues();
      // Undeclared exception!
      try { 
        mathValues0.startStage((StageConfiguration) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.myrobotlab.document.transformer.MathValues", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      MathValues mathValues0 = new MathValues();
      mathValues0.stopStage();
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      MathValues mathValues0 = new MathValues();
      mathValues0.flush();
  }
}
